File: backend/app/schemas/attachment.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from app.schemas.user import CamelModel, UserOut


class AttachmentOut(CamelModel):
    id: UUID
    original_filename: str
    #: The authenticated route that streams the bytes. `file_url` and
    #: `storage_key` are both withheld: the first was a public `/uploads/...`
    #: URL that defeated the authorization applied to reach this row, and the
    #: second discloses the server's storage layout for no client benefit.
    download_url: str
    mime_type: str
    file_size_bytes: int
    uploaded_by_id: UUID
    project_id: UUID
    entity_type: str
    entity_id: UUID
    uploaded_by: Optional[UserOut] = None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/attachment.ts
export type AttachmentEntityType = "TASK" | "TASK_REVIEW" | "ISSUE" | "SITE_REPORT" | "DESIGN_CHANGE";

export interface Attachment {
  id: string;
  originalFilename: string;
  /** Authenticated route that streams the bytes; see `useAuthedFile`. */
  downloadUrl: string;
  mimeType: string;
  fileSizeBytes: number;
  uploadedById: string;
  projectId: string;
  entityType: AttachmentEntityType;
  entityId: string;
  createdAt: string;
}
